Lesson35/src/App.js

```diff
@@ -0,0 +1,88 @@
+import { useReducer, useState } from 'react';
+import { List } from './modules/todo/List';
+import { Navbar } from './components/Navbar';
+import { Account } from './modules/user/Account';
+import { PerformanceState } from './modules/todo/PerformanceState';
+import { LogInForm } from './modules/user/LogInForm';
+import { todoData } from './data';
+import './App.css';
+
+const initialState = {
+  isLoggedInUser: false,
+  username: 'Guest',
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'LOGIN':
+      return {
+        ...state,
+        isLoggedInUser: true,
+        username: action.payload.username,
+      };
+    case 'LOGOUT':
+      return {
+        ...state,
+        isLoggedInUser: false,
+        username: 'Guest',
+      };
+    default:
+      return state;
+  }
+};
+
+export const App = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [todos, setTodos] = useState([]);
+  const [currentView, setCurrentView] = useState('list');
+  const handleLogin = (username) => {
+    dispatch({ type: 'LOGIN', payload: { username } });
+    setTodos(todoData);
+  };
+
+  const handleLogout = () => {
+    dispatch({ type: 'LOGOUT' });
+    setTodos([]);
+    setCurrentView('list');
+  };
+
+  const handleHomeClick = () => {
+    setCurrentView('list');
+  };
+
+  const handleAccountClick = () => {
+    setCurrentView('account');
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const totalTodos = todos.length;
+
+  return (
+    <>
+      <div className='container'>
+        <Navbar
+          isLoggedInUser={state.isLoggedInUser}
+          onSignOut={handleLogout}
```

Comment:
Don't pass this function as props to Navbar, get the dispatch function from context directly in the Navbar component.